internal/compress/gzip/gzip.go

```diff
@@ -0,0 +1,72 @@
+package gzip
+
+import (
+	"io"
+
+	"github.com/klauspost/compress/gzip"
+)
+
+// These constants are copied from the gzip package
+const (
+	NoCompression       = gzip.NoCompression
+	BestSpeed           = gzip.BestSpeed
+	BestCompression     = gzip.BestCompression
+	DefaultCompression  = gzip.DefaultCompression
+	ConstantCompression = gzip.ConstantCompression
+	HuffmanOnly         = gzip.HuffmanOnly
+)
+
+// ReaderWriter an interface to create Writer and Reader implementation.
+type ReaderWriter interface {
+	NewWriterLevel(w io.Writer, level int) (Writer, error)
```

Comment:
Should we apply functional option pattern here?
```suggestion
	NewWriterLevel(w io.Writer, ... wos WriterOption) (Writer, error)
```

Reply:
In this case, I think the writer option is not necessary.
Because it is necessary to write the exact same code as the option placed on the side of using this package.
https://github.com/vdaas/vald/blob/master/internal/compress/gzip_option.go#L46-L54

Reply:
and This gzip package is a wrapper for `klauspost/compress/gzip`. I would like a simpler interface with a similar interface.:thinking:
what do you think?

Reply:
I think it is an interface, so if we decided to use another gzip implementation, using functional option, we can easily to switch to the new implementation.
So I also suggest changing the function name to `NewWritter()`.
What do you think?